### PR TITLE
Support custom formatting function for errors

### PR DESCRIPTION
**Pull request by nfantone**
- Allow passing a `format` function to an `options` object to serialize errors.
- Unit tests.
- Expand documentation to showcase the feature.

### CODE REVIEW

**Comment by nfantone** [comment]
@jonathanong No love for this?

File: index.js [hunk 1]
```diff
@@ -1,12 +1,35 @@
...
   'type',
 ]
 
-module.exports = function () {
+var DEFAULTS = {
+  format: function(err) {
+    var obj = {}
+
+    // set all enumerable properties of error onto the object
+    Object.keys(err).forEach(function (key) {
```
Comment (jonathanong) [inline index.js]: can we just use a library for this. or `Object.assign()`?

**Comment by jonathanong** [comment]
@nfantone sorry. just gonna add you as a collab and give u npm publishing rights :)

File: index.js (hunk 1 above)
Comment (nfantone) [inline index.js]: Of course. But that bit was part of the original source. I just moved it here. And I wanted to make as little additions as possible.

**Comment by nfantone** [comment]
@jonathanong That would be swell!

**Pull request merged**